src/main/java/com/redhat/devtools/intellij/tektoncd/utils/VirtualFileHelper.java

```diff
@@ -35,14 +38,24 @@ public static void openVirtualFileInEditor(Project project, String namespace, St
         }
     }
 
-
     public static void createAndOpenVirtualFile(Project project, String namespace, String name, String content, String kind, ParentableNode<?> targetNode) {
         try {
-            VirtualFile vf = createTempFile(name, content);
+            VirtualFile vf;
+
+            //open TaskRun and PipelineRun in read only mode
+            if (KIND_PIPELINERUN.equals(kind) || KIND_TASKRUN.equals(kind)) {
+                vf = new LightVirtualFile(name, content);
+                vf.setWritable(false);
+            } else {
+                vf = createTempFile(name, content);
+                File fileToDelete = new File(vf.getPath());
```

Comment:
These two rows can be removed. To fix an issue with the cache we moved the deleteOnExit() inside the `createTempFile` method https://github.com/redhat-developer/intellij-tekton/blob/a52327b63b3131b87077e53f837b12d4b246d1c8/src/main/java/com/redhat/devtools/intellij/tektoncd/utils/VirtualFileHelper.java#L59

Reply:
Done